### PR TITLE
Added options to pin routes(##1x6691g)

### PR DESCRIPTION
**Pull request by disha1202**

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -33,9 +38,19 @@ export default defineComponent({
...
     IonToolbar
   },
+  computed: {
+    ...mapGetters({
+     pinnedPages: 'product/getPinnedPages'
+    })
+  },
+  methods:{
+    pinnedPage(){
+    console.log(this.pinnedPages[0]);
```
Comment (ymaheshwari1) [inline src/views/Home.vue]: Remove the console statement

File: src/views/ProductInventory.vue [hunk 2]
```diff
@@ -296,16 +298,39 @@ export default {
...
+    })
   },
   setup () {
     return {
       downloadOutline,
       folderOutline,
       swapVerticalOutline,
       sync,
-      filterOutline
+      filterOutline,
```
Comment (ymaheshwari1) [inline src/views/ProductInventory.vue]: Remove unwanted change of `,`

File: src/views/ProductInventory.vue [hunk 3]
```diff
@@ -296,16 +298,39 @@ export default {
...
+    return{
+      pagePinned: false,
+    }
+  },
+   methods:{
+    pinned(){
+      this.pagePinned = !this.pagePinned;
+      if(this.pagePinned){
+        this.pinnedPages.push("/product");
+        console.log(this.pinnedPages)
```
Comment (ymaheshwari1) [inline src/views/ProductInventory.vue]: Remove console statement

File: src/views/ProductInventory.vue [hunk 4]
```diff
@@ -296,16 +298,39 @@ export default {
...
+      }
+      else{
+       this.pinnedPages = "";
+      }
+      console.log(this.pinnedPages);
```
Comment (ymaheshwari1) [inline src/views/ProductInventory.vue]: Remove console statement

File: src/store/modules/product/actions.ts [hunk 5]
```diff
@@ -9,6 +9,10 @@ import emitter from '@/event-bus'
 
 
 const actions: ActionTree<ProductState, RootState> = {
+  pinnedpages({commit, state},payload){
+    commit(types.PINNED_PAGES, payload);
+    // return state.products.pinnedPages;
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts]: Remove commented code if not needed.

File: src/views/ProductInventory.vue [hunk 6]
```diff
@@ -296,16 +298,39 @@ export default {
...
+  },
+   methods:{
+    pinned(){
+      this.pagePinned = !this.pagePinned;
+      if(this.pagePinned){
+        this.pinnedPages.push("/product");
+        console.log(this.pinnedPages)
+      }
+      else{
+       this.pinnedPages = "";
```
Comment (ymaheshwari1) [inline src/views/ProductInventory.vue]: Why we are assigning empty string to pinnedPages property?

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]
I think we should create the state of pinned pages in the user module and not the product module, as the pages that are pinned will be associated with a user. WDYT?

**Comment by disha1202** [comment]
> I think we should create the state of pinned pages in the user module and not the product module, as the pages that are pinned will be associated with a user. WDYT?

Yes sir, I also think that the pinned pages state should be in user module

File: src/views/ProductInventory.vue (hunk 6 above)
Comment (disha1202) [reply src/views/ProductInventory.vue]: Previously we were using an object instaed of an array, so to empty a slot I assigned empty string to pinned pages. Now I have updated the code to remove that element from array.

File: src/store/modules/product/actions.ts (hunk 5 above)
Comment (disha1202) [reply src/store/modules/product/actions.ts]: Removed.

File: src/views/ProductInventory.vue (hunk 4 above)
Comment (disha1202) [reply src/views/ProductInventory.vue]: Remved

File: src/views/ProductInventory.vue (hunk 3 above)
Comment (disha1202) [reply src/views/ProductInventory.vue]: Removed

File: src/views/ProductInventory.vue (hunk 2 above)
Comment (disha1202) [reply src/views/ProductInventory.vue]: Done

File: src/views/Home.vue (hunk 1 above)
Comment (disha1202) [reply src/views/Home.vue]: Removed

File: src/store/modules/user/actions.ts [hunk 7]
```diff
@@ -11,6 +11,11 @@ import "moment-timezone";
 
 const actions: ActionTree<UserState, RootState> = {
 
+  pinnedpages({ commit },payload) {
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts]: Use camelCase for naming actions.

File: src/store/modules/user/mutation-types.ts [hunk 8]
```diff
@@ -2,4 +2,5 @@ export const SN_USER = 'user'
 export const USER_TOKEN_CHANGED = SN_USER + '/TOKEN_CHANGED'
 export const USER_END_SESSION = SN_USER + '/END_SESSION'
 export const USER_INFO_UPDATED = SN_USER + '/INFO_UPDATED'
-export const USER_CURRENT_FACILITY_UPDATED = SN_USER + '/CURRENT_FACILITY_UPDATED'
\ No newline at end of file
+export const USER_CURRENT_FACILITY_UPDATED = SN_USER + '/CURRENT_FACILITY_UPDATED'
+export const PINNED_PAGES = 'pinned'
```
Comment (ymaheshwari1) [inline src/store/modules/user/mutation-types.ts]: Use the same way to define the mutation type as defined for the other mutations.

File: src/store/modules/user/UserState.ts [hunk 9]
```diff
@@ -2,4 +2,5 @@ export default interface UserState {
     token: string;
     current: object | null;
     currentFacility: object;
+    pinnedPages:[];
```
Comment (ymaheshwari1) [inline src/store/modules/user/UserState.ts]: Define the type like `Array<any>`

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 7 above)
Comment (disha1202) [reply src/store/modules/user/actions.ts]: Done sir

File: src/store/modules/user/mutation-types.ts (hunk 8 above)
Comment (disha1202) [reply src/store/modules/user/mutation-types.ts]: Done sir

File: src/store/modules/user/UserState.ts (hunk 9 above)
Comment (disha1202) [reply src/store/modules/user/UserState.ts]: Done sir

File: src/views/Home.vue [hunk 10]
```diff
@@ -33,9 +38,18 @@ export default defineComponent({
...
     IonToolbar
   },
+  computed: {
+    ...mapGetters({
+     pinnedPages: 'user/getPinnedPages'
+    })
+  },
+  methods:{
+    pinnedPage(){
+    this.$router.push(this.pinnedPages[0]);
```
Comment (ymaheshwari1) [inline src/views/Home.vue:48]: Improve indenation

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Comment by Dhiraj1405** [comment]
Not priority for now. Hence, closing this pull request.

**Pull request closed without merging**